healthyregions/SDOHPlace
# fix: refactored lib folder patterns surrounding posts + showcases

## Problem
There is quite a bit of copy-pasted code between the Posts + Showcases

We would like a better pattern for adding new categories to avoid copy-pasting as much in the future

Redo of #312 

Fixes #302 

## Approach
* fix: restructured PostLayout + Showcase into `src/components/Layout`
* fix: refactor PostLayout + ShowcaseLayout slightly to more easily include new categories in the future (e.g. Guides, see #317)
* fix: refactor lib folder to cut down on some of the copy-pasted bits - maintain separate implementations for these resources
    * NOTE: After cleaning up this code, the Showcases are no longer sorted by Date - this will be fixed by #303 

## How to Test
1. Navigate to the list of "Project News" (PostList / PostItem): http://localhost:3000/news
    * You should see no changes compared to production: https://sdohplace.org/news
2. Choose an article (Layout -> PostLayout): http://localhost:3000/news/finding-healthy-food-connecticut
    * You should see no changes compared to production: https://sdohplace.org/news/finding-healthy-food-connecticut
    * NOTE: Images appear differently here, but these are not style changes - production appears to use different screenshot paths than dev
    * NOTE: Copyright symbol (at the bottom) has moved to the left a few pixels - I could not track down why, but this is barely noticeable
3. Navigate to the list of "Fellow Showcases" (ShowcaseList / ShowcaseItem): http://localhost:3000/showcase
    * You should see no changes compared to production: https://sdohplace.org/showcase
    * NOTE: Images appear differently here, but these are not style changes - production appears to use different screenshot paths than dev
4. Choose a showcase (Layout -> ShowcaseLayout): http://localhost:3000/showcase/windham-willmantic-conn-food-access-map
    * You should see no changes compared to production: https://sdohplace.org/showcase/windham-willmantic-conn-food-access-map
    * NOTE: Images appear differently here, but these are not style changes - production appears to use different screenshot paths than dev
    * NOTE: Copyright symbol (at the bottom) has moved to the left a few pixels - I could not track down why, but this is barely noticeable
